### PR TITLE
DOCS-5096, DOCS-6360 Pipeline Visibility Update

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Add a feature support table that explains what features/telemetry is supported for various CI providers for Pipeline Visibility.

Chat with Kassen on Slack, also https://datadoghq.atlassian.net/browse/DOCS-5096.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Comment by kassenq** [comment]
IMO we should remove the job dependency information row because we only support one provider currently, and it comes across as a shortcoming of our product

File: content/en/continuous_integration/pipelines/_index.md [hunk 1]
```diff
@@ -26,6 +26,24 @@ cascade:
...
+| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the **Logs** tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
+| {{< ci-details title="Infrastructure metric correlation" >}}The ability to display host-level information for the Datadog Agent or runners running jobs or pipelines.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |  |
+| {{< ci-details title="Custom spans for traced commands using datadog-ci" >}}The ability to send command-level events to CI Visibility so the events can be incorporated in the pipeline flame graph. You can then query and analyze <a href="https://docs.datadoghq.com/continuous_integration/pipelines/custom_commands/">these events</a>. {{< /ci-details >}} | {{< X >}} |  | {{< X >}} |  | {{< X >}} |  |  |  |
+| {{< ci-details title="Custom predefined tags" >}}The ability to set static pipeline tags that do not change between executions.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:41]:
```suggestion
| {{< ci-details title="Custom predefined tags" >}}The ability to set static pipeline tags that do not change between executions in a CI provider.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 2]
```diff
@@ -26,6 +26,24 @@ cascade:
 
 [Pipeline Visibility][1] provides a pipeline-first view into your CI health by displaying important metrics and results from your pipelines. It can help you investigate performance problems and test failures that concern you the most because you work on the related code, not because you maintain the pipelines they are run in.
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:27]:
```suggestion
[Pipeline Visibility][1] provides a pipeline-first view into your CI health by displaying important metrics and results from your pipelines. It helps you troubleshoot pipeline failures, address performance bottlenecks, and track CI performance and reliability over time.
```

**Review by drichards-87 — APPROVED** [review]

File: content/en/continuous_integration/pipelines/_index.md [hunk 3]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Manual steps" >}}The ability to identify when there is a job with a manual approval phase in the overall pipeline, which is split into two pipelines: <br>- From the beginning of the pipeline to when the manual approval action starts and the pipeline becomes <code>Blocked</code>.<br>- From the end of the manual approval action to the end of the pipeline.<br>{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  | {{< X >}} |  |
+| {{< ci-details title="Queue time" >}}The ability to add the time for which a job (or pipeline depending on the CI provider) was in the queue before it executes.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the **Logs** tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:37]:
```suggestion
| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the <b>Logs</b> tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 4]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Manual steps" >}}The ability to identify when there is a job with a manual approval phase in the overall pipeline, which is split into two pipelines: <br>- From the beginning of the pipeline to when the manual approval action starts and the pipeline becomes <code>Blocked</code>.<br>- From the end of the manual approval action to the end of the pipeline.<br>{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  | {{< X >}} |  |
+| {{< ci-details title="Queue time" >}}The ability to add the time for which a job (or pipeline depending on the CI provider) was in the queue before it executes.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the <b>Logs</b> tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:37]:
```suggestion
| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the <strong>Logs</strong> tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/pipelines/_index.md [hunk 5]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Partial retries" >}}The ability to identify partial retries (for example, when only a subset of jobs were retried) in CI providers.{{< /ci-details >}} |  | {{< X >}} |  | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Manual steps" >}}The ability to identify when there is a job with a manual approval phase in the overall pipeline, which is split into two pipelines: <br>- From the beginning of the pipeline to when the manual approval action starts and the pipeline becomes <code>Blocked</code>.<br>- From the end of the manual approval action to the end of the pipeline.<br>{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  | {{< X >}} |  |
+| {{< ci-details title="Queue time" >}}The ability to add the time for which a job (or pipeline depending on the CI provider) was in the queue before it executes.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:36]: Identification of amount of time for which a pipeline or job was in the queue before execution

File: content/en/continuous_integration/pipelines/_index.md [hunk 6]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Logs correlation" >}}The ability to retrieve pipeline or job logs from the CI provider. Logs are displayed on the <strong>Logs</strong> tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:37]: Retrieval of pipeline or job logs from the CI provider

File: content/en/continuous_integration/pipelines/_index.md [hunk 7]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Infrastructure metric correlation" >}}The ability to display host-level information for the Datadog Agent, pipelines, or job runners.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:38]: Correlation of host-level information for the Datadog Agent, CI pipelines, or job runners to CI pipeline execution data

File: content/en/continuous_integration/pipelines/_index.md [hunk 8]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Custom spans for traced commands using datadog-ci" >}}The ability to send command-level events to CI Visibility so the events can be incorporated in the pipeline flame graph. You can then query and analyze <a href="https://docs.datadoghq.com/continuous_integration/pipelines/custom_commands/">these events</a>. {{< /ci-details >}} | {{< X >}} |  | {{< X >}} |  | {{< X >}} |  |  |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:39]: Support for sending command-level events to CI Visibility for incorporation into pipeline flame graph visualization. ...

Comment (kassenq) [reply content/en/continuous_integration/pipelines/_index.md:39]: The "..." means keep the rest of it the same! :)

File: content/en/continuous_integration/pipelines/_index.md [hunk 9]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Custom predefined tags" >}}The ability to set static pipeline tags that do not change between executions in a CI provider.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:40]: Support for setting static pipeline tags in the CI provider that do not change between executions

File: content/en/continuous_integration/pipelines/_index.md [hunk 10]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Custom tags and metrics at runtime" >}}The ability to add <a href="https://docs.datadoghq.com/continuous_integration/pipelines/custom_tags_and_metrics/">user-defined text and numerical tags</a> to pipelines and jobs in CI Visibility.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:41]: Support for adding user-defined text and numerical tags to pipelines and jobs in CI Visibility


(keep hyperlink the same)

File: content/en/continuous_integration/pipelines/_index.md [hunk 11]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Parameters" >}}The ability to add the custom pipeline parameters that users set (for example, <code>DYNAMICS_IS_CHILD:true</code>). Users can search using these parameters in the Datadog UI to find all the events with a specific parameter.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  |  |  | {{< X >}} |  |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:42]: Support for adding custom pipeline parameters ... You can then search using these parameters in the Pipeline Executions page to find all events with a specific parameter

File: content/en/continuous_integration/pipelines/_index.md [hunk 12]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+| {{< ci-details title="Pipeline failure reason" >}}The ability to add the reason for a failure of a pipeline or a job.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  |  |  | {{< X >}} | {{< X >}} |
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/_index.md:43]: Identification of specific reason behind pipeline or job failure

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/pipelines/_index.md (hunk 12 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:43]:
```suggestion
| {{< ci-details title="Pipeline failure reason" >}}Identification of a specific reason behind a pipeline or job failure.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  |  |  | {{< X >}} | {{< X >}} |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 11 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:42]:
```suggestion
| {{< ci-details title="Parameters" >}}Support for adding custom pipeline parameters that users set (for example, <code>DYNAMICS_IS_CHILD:true</code>). You can then search using these parameters in the <a href="https://docs.datadoghq.com/continuous_integration/explorer/?tab=pipelineexecutions">CI Visibility Explorer</a> to find all events with a specific parameter.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  |  |  | {{< X >}} |  |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 10 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:41]:
```suggestion
| {{< ci-details title="Custom tags and metrics at runtime" >}}Support for adding <a href="https://docs.datadoghq.com/continuous_integration/pipelines/custom_tags_and_metrics/">user-defined text and numerical tags</a> to pipelines and jobs in CI Visibility.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 9 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:40]:
```suggestion
| {{< ci-details title="Custom predefined tags" >}}Support for setting static pipeline tags in the CI provider that do not change between executions.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 8 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:39]:
```suggestion
| {{< ci-details title="Custom spans for traced commands using datadog-ci" >}}Support for sending command-level events to CI Visibility to be incorporated into pipeline flame graph visualization. You can then query and analyze <a href="https://docs.datadoghq.com/continuous_integration/pipelines/custom_commands/">these events</a>. {{< /ci-details >}} | {{< X >}} |  | {{< X >}} |  | {{< X >}} |  |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 6 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:37]:
```suggestion
| {{< ci-details title="Logs correlation" >}}Retrieval of pipeline or job logs from the CI provider. Logs are displayed on the <strong>Logs</strong> tab in the Pipeline Execution view.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  | {{< X >}} |  |  |  |
```

File: content/en/continuous_integration/pipelines/_index.md (hunk 5 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:36]:
```suggestion
| {{< ci-details title="Queue time" >}}Identification of the amount of time for which a pipeline or job was in the queue before execution.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |  | {{< X >}} | {{< X >}} |
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 13]
```diff
@@ -24,7 +24,23 @@ cascade:
...
+
+### Supported features
+
+|  | Jenkins | GitLab | CircleCI | Buildkite | GitHub Actions | Azure Pipelines | Codefresh | TeamCity |
+|---|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| {{< ci-details title="Pipeline trace visualization" >}}The ability to do pipeline execution visualization with associated tracing.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Partial retries" >}}The ability to identify partial retries (for example, when only a subset of jobs were retried) in CI providers.{{< /ci-details >}} |  | {{< X >}} |  | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Manual steps" >}}The ability to identify when there is a job with a manual approval phase in the overall pipeline, which is split into two pipelines: <br>- From the beginning of the pipeline to when the manual approval action starts and the pipeline becomes <code>Blocked</code>.<br>- From the end of the manual approval action to the end of the pipeline.<br>{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  | {{< X >}} |  |
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:35]:
```suggestion
| {{< ci-details title="Manual steps" >}}Identification of when there is a job with a manual approval phase in the overall pipeline.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  | {{< X >}} |  |  | {{< X >}} |  |
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 14]
```diff
@@ -24,7 +24,23 @@ cascade:
...
 ## Overview
 
-[Pipeline Visibility][1] provides a pipeline-first view into your CI health by displaying important metrics and results from your pipelines. It can help you investigate performance problems and test failures that concern you the most because you work on the related code, not because you maintain the pipelines they are run in.
+[Pipeline Visibility][1] provides a pipeline-first view into your CI health by displaying important metrics and results from your pipelines. It helps you troubleshoot pipeline failures, address performance bottlenecks, and track CI performance and reliability over time.
+
+### Supported features
+
+|  | Jenkins | GitLab | CircleCI | Buildkite | GitHub Actions | Azure Pipelines | Codefresh | TeamCity |
+|---|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| {{< ci-details title="Pipeline trace visualization" >}}The ability to do pipeline execution visualization with associated tracing.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/pipelines/_index.md:33]:
```suggestion
| {{< ci-details title="Pipeline trace visualization" >}}Visualization of pipeline executions with associated tracing.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
```

**Review by kassenq — APPROVED** [review]

**Pull request merged**